Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide/Internal.hs

```diff
@@ -62,10 +68,27 @@ cancelRideImpl rideId rideEndedBy bookingCReason = do
     when (bookingCReason.source == SBCR.ByDriver) $
       DS.driverScoreEventHandler ride.merchantOperatingCityId DST.OnDriverCancellation {merchantId = merchantId, driverId = driver.id, rideFare = Just booking.estimatedFare, currency = booking.currency, distanceUnit = booking.distanceUnit}
     Notify.notifyOnCancel ride.merchantOperatingCityId booking driver bookingCReason.source
-
   fork "cancelRide/ReAllocate - Notify BAP" $ do
     isReallocated <- reAllocateBookingIfPossible isValueAddNP False merchant booking ride driver vehicle bookingCReason
-    unless isReallocated $ BP.sendBookingCancelledUpdateToBAP booking merchant bookingCReason.source
+    unless isReallocated $ BP.sendBookingCancelledUpdateToBAP booking merchant bookingCReason.source noShowCharges
+
+calculateNoShowCharges :: SRB.Booking -> DRide.Ride -> Flow (Maybe PriceAPIEntity)
+calculateNoShowCharges booking ride = do
+  mbFarePolicy <- SFP.getFarePolicyByEstOrQuoteIdWithoutFallback booking.quoteId
```

Comment:
Can we fetch cancellation when we are fetching FarePolicy only and make it part of FarePolicy Domain type?

Reply:
Done, made cancellation fare policy type of full fare policy domain type.